### PR TITLE
[Woo POS] Floating control for menu and reader status

### LINKED ISSUE #13338: [Woo POS] M2: Floating control for menu and reader status
p91TBi-bHy-p2

Image

### PR DESCRIPTION
**Pull request by bozidarsevo**
Closes: #13338

## Description

Added basic implementation of floating control for POS menu and reader status. p91TBi-bHy-p2

Floating control consists of 2 parts:
- "⋯" button that shows menu items "Get Support" and "Exit POS"
- Card reader status info view that shows card read connection status and if reader is not connected it is a button which starts the connection process

**Floating control overlaps some of the UI still and that will be updated as we update other UI parts of the UI to M2 design**

## Testing information

- Enter POS
- Use floating control to connect to card reader by tapping "Connect your reader" button
- Make sure that the reader is connected
- Add few products to cart
- Tap Checkout
- Finish the payment
- Tap "⋯" button in floating control and tap "Exit POS"

## Demo video

https://github.com/user-attachments/assets/41f5b03d-2e22-477a-a008-4f8c0df1c618

---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/215">19.7</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift [hunk 1]
```diff
@@ -72,11 +76,24 @@ struct PointOfSaleDashboardView: View {
     }
 }
 
+struct FloatingControlAreaSizeKey: EnvironmentKey {
```
Comment (bozidarsevo) [inline WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift:79]: Added support for environment variable for easier passing of the floating control area size

**Review by soundsokay — COMMENTED** [review]
I tested on a physical iPad Pro (12.9-inch, 3rd generation) running iPadOS Version 16.6.1.

### Test Results

I went through the steps in the Testing Information section in order. Each step worked, however I noted a visual discrepancy when compared to the design and a screenshot from an Android demo.

### 3-dot menu 

When the menu is open the 3-dot menu button below is only faintly visible:
Screenshot 2024-07-23 at 11 30 52 PM

This differs from the Figma design and a screenshot from an Android demo:

**Figma design**
3-dot menu opened - Design - Screenshot 2024-07-23 at 9 19 53 PM

**Android demo**
3-dot menu opened - Android demo- Screenshot 2024-07-23 at 9 19 53 PM

**Can the 3-dot button be made more visible when the menu is open to match the design?**

**Comment by bozidarsevo** [comment]
> Can the 3-dot button be made more visible when the menu is open to match the design?

Updated in latest commit. Regarding the label of the SwiftUI `Menu` component, I can create a custom menuStyle for it if we want to change the color of the three dot button when it is selected, but I was thinking maybe to leave it to the system to update it. Or would a custom menuStyle be better, and be 1on1 with Figma? If I create a custom menuStyle we need to handle the dark mode in the future

Simulator Screenshot - iPad Pro (11-inch) (4th generation) 17 2 - 2024-07-24 at 14 01 38

File: WooCommerce/Classes/POS/Presentation/POSFloatingControlView.swift [hunk 2]
```diff
@@ -34,9 +34,9 @@ struct POSFloatingControlView: View {
                         .font(Constants.ellipsisFont)
                 }
                 .frame(width: Constants.size, height: Constants.size)
-                .background(Color.white)
-                .cornerRadius(Constants.cornerRadius)
             }
+            .background(Color.white)
```
Comment (soundsokay) [inline WooCommerce/Classes/POS/Presentation/POSFloatingControlView.swift:38]: 👍

**Review by soundsokay — APPROVED** [review]
I'm now seeing better visibility of the 3-dot menu.
adjusted UI for opened 3-dot menu - Screenshot 2024-07-24 at 11 34 27 AM

Let's leave the button styling to the system, at least for now.

Suggest filing a new issue for revisiting this later including whether or not to add a custom SwiftUI MenuStyle to exactly match the Figma design.

**Comment by staskus** [comment]
> Let's leave the button styling to the system, at least for now.

👍 Agree, fits to what we talked about.

**Pull request merged**